src/dbnode/storage/series/reader.go

```diff
@@ -208,7 +217,15 @@ func (r Reader) fetchBlocksWithBlocksMapAndBuffer(
 	nsCtx namespace.Context,
 ) ([]block.FetchBlockResult, error) {
 	var (
-		// TODO(r): pool these results arrays
+		// Two-dimensional slice such that the first dimension is unique by blockstart
+		// and the second dimension is blocks of data for that blockstart (not necessarily
+		// in chronological order).
```

Comment:
Seems like a copy pasta comment from above that doesn't follow here.

Reply:
I refactored it to be more clear, take another look